pallets/ajuna-awesome-avatars/src/tests.rs

```diff
@@ -385,147 +388,156 @@ mod season {
-					}
-				);
-			}
+				// Season 1 early start (block 2..3)
+				for n in season_1_schedule.early_start..season_1_schedule.start {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 1,
+							early: true,
+							active: false,
+							early_ended: false,
+							max_tier_avatars: 0
+						}
+					);
+				}
+				// Season 1 start (block 3..4)
+				for n in season_1_schedule.start..season_2_schedule.early_start {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 1,
+							early: false,
+							active: true,
+							early_ended: false,
+							max_tier_avatars: 0
+						}
+					);
+				}
 
-			// No further seasons exist
-			assert!(Seasons::<Test>::get(CurrentSeasonStatus::<Test>::get().season_id).is_none());
-		})
+				// Season 2 early start (block 5..6)
+				for n in season_2_schedule.early_start..season_2_schedule.start {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 2,
+							early: true,
+							active: false,
+							early_ended: false,
+							max_tier_avatars: 0
+						}
+					);
+				}
+				// Season 2 start (block 7..9)
+				for n in season_2_schedule.start..season_2_schedule.end {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 2,
+							early: false,
+							active: true,
+							early_ended: false,
+							max_tier_avatars: 0
+						}
+					);
+				}
+				// Season 2 end (block 10..22)
+				for n in (season_2_schedule.end + 1)..season_3_schedule.early_start {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 3,
+							early: false,
+							active: false,
+							early_ended: false,
+							max_tier_avatars: 0
+						}
+					);
+				}
+
+				// Season 3 early start (block 23..36)
+				for n in season_3_schedule.early_start..season_3_schedule.start {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 3,
+							early: true,
+							active: false,
+							early_ended: false,
+							max_tier_avatars: 0
+						}
+					);
+				}
+				// Season 3 start (block 37..53)
+				for n in season_3_schedule.start..=season_3_schedule.end {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 3,
+							early: false,
+							active: true,
+							early_ended: false,
+							max_tier_avatars: 0
+						}
+					);
+				}
+				// Season 3 end (block 54..63)
+				for n in (season_3_schedule.end + 1)..=(season_3_schedule.end + 10) {
+					run_to_block(n);
+					assert_eq!(
+						CurrentSeasonStatus::<Test>::get(),
+						SeasonStatus {
+							season_id: 4,
+							early: false,
+							active: false,
+							early_ended: false,
```

Comment:
To me, it seems that the three options are mutually exclusive. This makes me think that an `enum SeaonTimeline { inactive, early, active, earlyEnded }` is a better choice, as it is less error-prone.